packages/framework/build.xml

```diff
@@ -0,0 +1,1029 @@
+    <target name="build-deploy-part-1-db-independent" depends="generate-build-version,clean,composer,redis-check,npm,dirs-create,assets" description="First part of application build for production preserving your DB (can be run without maintenance page)."/>
+    <target name="build-deploy-part-2-db-dependent" depends="db-migrations,create-domains-data,generate-friendly-urls,replace-domains-urls,grunt,error-pages-generate,warmup,clean-redis-old" description="Second part of application build for production preserving your DB (must be run with maintenance page when containing DB migrations)."/>
+    <target name="build-new" depends="wipe,generate-build-version,composer,redis-check,npm,dirs-create,assets,db-rebuild,grunt,error-pages-generate,warmup,product-search-recreate-structure,clean-redis-old" description="Builds application for production with clean DB (with base data only)."/>
+    <target name="build-demo" depends="wipe,generate-build-version,composer,redis-check,npm,dirs-create,assets,db-demo,grunt,error-pages-generate,warmup,product-search-recreate-structure,product-search-export-products,clean-redis-old" description="Builds application for production with clean demo DB."/>
+    <target name="db-demo" depends="db-wipe-public-schema,db-import-basic-structure,db-migrations,create-domains-data,db-fixtures-demo,load-plugin-demo-data,generate-friendly-urls,replace-domains-urls" description="Creates DB and fills it with demo data"/>
+    <target name="db-rebuild" depends="db-wipe-public-schema,db-import-basic-structure,db-migrations,create-domains-data,generate-friendly-urls,replace-domains-urls" description="Drops all data in database and creates a new one with base data only."/>
+    <target name="product-search-recreate-structure" depends="product-search-delete-structure,product-search-create-structure" description="Recreates structure for searching via elasticsearch (deletes existing structure and creates new one)"/>
+
+    <target name="assets" description="Installs web assets from external bundles into a public web directory.">
+        <exec executable="${path.php.executable}" passthru="true" checkreturn="true" output="${dev.null}">
+            <arg value="${path.bin-console}"/>
+            <arg value="assetic:dump"/>
+        </exec>
+
+        <delete failonerror="false" includeemptydirs="true">
+            <fileset dir="${path.root}/web/bundles/">
+                <exclude name="/"/>
+            </fileset>
+        </delete>
+        <exec executable="${path.php.executable}" passthru="true" checkreturn="true">
+            <arg value="${path.bin-console}"/>
+            <arg value="assets:install"/>
+            <arg value="${path.root}/web/"/>
+        </exec>
+    </target>
+
+    <target name="clean" description="Cleans up directories with cache and scripts which are generated on demand.">
```

Comment:
`clean` should not depend on `clean-redis` as it needs the DIC - it cannot be executed before `composer(-dev)`

Reply:
So why you can't just use `clean-var` in places where you don't have DIC yet. 
I read https://github.com/shopsys/shopsys/issues/742 issue again and I think, that this PR doesn't close that issue because there are no rules for those targets and specific `clean` were mentioned in that issue. 

I would like to use just one command to clean cache after I change branch instead of two every time.

Reply:
@henzigo I see the mention of `clean` in #742 as an example. There are some rules now (see [Guidelines for Phing Targets](https://github.com/shopsys/shopsys/blob/b5405eadbc22a9a8e3436c179988e83aa81c5bfb/docs/contributing/guidelines-for-phing-targets.md)) and the problem with lots of different targets was solved by hiding a lot of them.

The `clean` target is depended on from a lot of targets, scripts and places and I don't really feel confident to change it significantly as it may mean a lot of work for people during upgrade. Maybe a different approach by creating a cleaning command just for switching branches would be better, but I'm not sure how to name it (`build-dev-quicker`? `refresh`? ...?) or what it should include:
```xml
<target name="???" depends="clean,composer-dev,clean-redis" description="Cleans the application from autogenerated files and cache and installs depedencies"/>
```

(The `compose install` runs pretty quickly so I would definitely include it, apart from that, I'm not sure... Maybe DB migrations? Maybe Grunt?)

Reply:
I understand you, but you moved all commands to framework, so you don't need to deal with upgrade notes. I absolutely want some target to clean all caches. If you do not want to use `clean`, you can create `clean-all` or something. Probably clean cache is most used command and this time and nobody wants to do it with two commands.

Secondly, there are some guidelines for phing targets, but I'm missing some categories with description and common sence like:
1. Build: **build**-dev, **build**-new
2. Test: **test**-unit, **test**-smoke ....
3. Clean: **clean**-var, **clean**-redis

Now that targets doesn't make any sense and I will be glad if you do with it something.

Reply:
`clean-all` would have to run `clean-styles` as well, which would require you to run `grunt` afterwards...

I hope that the categories are self-evident, that they don't need any further explanation - you disagree?

Reply:
How about adding this?

```xml
<target name="refresh" depends="clean,composer-dev,clean-redis,db-migrations,grunt" description="Refreshes the application cache and generated JS / CSS files autogenerated files and installs DB migrations if needed"/>
```

It's pretty quick, runs for 15 seconds on my machine and should cover most of the things needed when switching between branches...

Reply:
After discussion I've added `clean-cache` target that cleans both filecache and Redis cache in 6581e2f8d97c3e68e57a0cb8958f4b16f6f54c7c.